RoslynPluginGenerator/ZipExtensions.cs

```diff
@@ -53,7 +50,7 @@ public static void CreateFromDirectory(string sourceDirectoryName, string destin
             {
                 foreach (string file in files.Where(f => fileInclusionPredicate(f)))
                 {
-                    archive.CreateEntryFromFile(file, file.Substring(pathPrefixLength), CompressionLevel.Optimal);
+                    archive.CreateEntryFromFile(file, file.Substring(pathPrefixLength).Replace("\\", "/"), CompressionLevel.Optimal);
```

Comment:
while this should work, I guess it would be better to use the `System.IO.Path.DirectorySeparatorChar` as replacement to make it fit to the operating system
```suggestion
archive.CreateEntryFromFile(file, file.Substring(pathPrefixLength).Replace("\\", Path.DirectorySeparatorChar), CompressionLevel.Optimal);
```

Reply:
That does not fix the issue. If the archive is created on Windows and then extracted on Linux we will get the same issue.
On the most recent operating systems, forward slash is a valid directory separator; I prefer to keep the hardcoded `/` for safety.

Reply:
Based on my test, this is not working when using `Path.DirectorySeparatorChar`.
When running the RoslynPluginGenerator on windows, it would be equivalent to do `.Replace("\\", "\\")` or do nothing.
Then when doing a `dotnet sonarscanner begin` in Linux (WSL in my case) the extracted pathes are not correct:
![image](https://github.com/SonarSource/sonarqube-roslyn-sdk/assets/126669183/c33a9c0a-7afe-48f1-8464-460dd8deeabc)

Reply:
I don't want to be too pedantic but it's working as expected on Linux (including WSL) and Windows.

Linux (WSL)
![image](https://github.com/SonarSource/sonarqube-roslyn-sdk/assets/3630638/f8735c66-53e0-4e64-9c53-c8f08d47a393)

Windows
![image](https://github.com/SonarSource/sonarqube-roslyn-sdk/assets/3630638/8402d9d0-5aea-4800-820e-1ce88e9533ee)

I am not one to merge this pull request, maybe someone else should decide what to do. I guess `/` works for pretty much every operating system out there but I am also confident that the directory separator char works as expected.

Reply:
I am not saying that `Path.DirectorySeparatorChar` is not doing what it is expected to do.
I am saying that using `Path.DirectorySeparatorChar` to solve the path separator issue in the jar (zip) file generated by the RoslynPluginGenerator is not a good solution.

This fix is only about the creation of the archive and not the extraction.
If it were about the extraction, I would agree that `Path.DirectorySeparatorChar` would be a nicer solution.

We need to ensure that all entry names in the archive **do not contain** any backslashes.

Reply:
`/` it is, to close this